src/components/selection-field.jsx

```diff
@@ -67,7 +67,11 @@ export default function SelectionField({
 
   let positioning = {};
   if (selfRef && selfRef.current) {
-    positioning = { left: `${selfRef.current.getBoundingClientRect().left}px` };
+    const boundingRect = selfRef.current.getBoundingClientRect();
```

Comment:
I'm thinking, maybe we can move this logic into a new hook, `useVisible`, that would do the logic needed to check if it's visible?

Hook example:

```js
const selfRef = useRef(null);
const isVisible = useVisible(selfRef.current);
const [showFilterbox, setShowFilterbox] = useState(false);
useEffect(() => setShowFilterbox(isVisible), [isVisible]);

[rest of logic]
```

Hook logic could probably avoid the `maxWidth` prop also then, by using:

```js
const rect = el.getBoundingClientRect();
const elem = document.elementFromPoint(rect.left, rect.top);
const isVisible = selfRef.current === elem;
```

Sorry, it's 6am here so if this doesn't make sense, ignore :)